examples/nlp/token_classification/punctuate_capitalize_infer.py

```diff
@@ -0,0 +1,739 @@
+    ]:
+        inp_ids, segment_ids, inp_mask, st_mask, n_preceding, query_ids, is_first, is_last = zip(*batch)
+        return (
+            pad_sequence([torch.tensor(x) for x in inp_ids], batch_first=True, padding_value=0),
+            pad_sequence([torch.tensor(x) for x in segment_ids], batch_first=True, padding_value=0),
+            pad_sequence([torch.tensor(x) for x in inp_mask], batch_first=True, padding_value=0),
+            pad_sequence([torch.tensor(x) for x in st_mask], batch_first=True, padding_value=0),
+            n_preceding,
+            query_ids,
+            is_first,
+            is_last,
+        )
+
+    def __getitem__(self, idx: int) -> Tuple[np.nddarray, np.ndarray, np.ndarray, np.ndarray, int, int, bool, bool]:
+        return (
+            np.array(self.all_input_ids[idx]),
+            np.array(self.all_segment_ids[idx]),
+            np.array(self.all_input_mask[idx], dtype=np.float32),
+            np.array(self.all_subtokens_mask[idx]),
+            self.all_quantities_of_preceding_words[idx],
+            self.all_query_ids[idx],
+            self.all_is_first[idx],
+            self.all_is_last[idx],
+        )
+
+
+def setup_infer_long_dataloader(
+    model: PunctuationCapitalizationModel,
+    queries: List[str],
+    batch_size: int,
+    max_seq_length: int,
+    step: int,
+    margin: int,
+) -> torch.utils.data.DataLoader:
+    """
+    Setup function for a infer data loader.
+
+    Args:
+        model: a ``PunctuationCapitalizationModel`` instance for which data loader is created.
+        queries: lower cased text without punctuation
+        batch_size: batch size to use during inference
+        max_seq_length: length of segments into which queries are split. ``max_seq_length`` includes ``[CLS]`` and
+            ``[SEP]`` so every segment contains at most ``max_seq_length-2`` tokens from input a query.
+        step: number of tokens by which a segment is offset to a previous segment. Parameter ``step`` cannot be greater
+            than ``max_seq_length-2``.
+        margin: number of tokens near the edge of a segment which label probabilities are not used in final prediction
+            computation.
+    Returns:
+        A pytorch DataLoader.
+    """
+    if max_seq_length is None:
+        max_seq_length = model._cfg.dataset.max_seq_length
+    if step is None:
+        step = model._cfg.dataset.step
+    if margin is None:
+        margin = model._cfg.dataset.margin
+
+    dataset = BertPunctuationCapitalizationInferLongDataset(
+        tokenizer=model.tokenizer, queries=queries, max_seq_length=max_seq_length, step=step, margin=margin
+    )
+    return torch.utils.data.DataLoader(
+        dataset=dataset,
+        collate_fn=dataset.collate_fn,
+        batch_size=batch_size,
+        shuffle=False,
+        num_workers=model._cfg.dataset.num_workers,
+        pin_memory=model._cfg.dataset.pin_memory,
+        drop_last=False,
+    )
+
+
+def _move_acc_probs_to_token_preds(
```

Comment:
move to the model class or create a utils file inside https://github.com/NVIDIA/NeMo/tree/main/nemo/collections/nlp/models/token_classification